### PR TITLE
Updated Models and related functions

### PR DESCRIPTION
**Pull request by epadams**
- Added other components to models
- Updated ToBaseProduct and GetRequiredPropertiesForCategories with new models
- Updated ApplicationDbContext to handle new models

### CODE REVIEW

**Comment by epadams** [comment]
Side discussion, I noticed a lot of warning errors since many of our properties (notably strings) can be nullable. Should we go ahead and update these?

**Comment by evan-scales** [comment]
> Side discussion, I noticed a lot of warning errors since many of our properties (notably strings) can be nullable. Should we go ahead and update these?

You can set them equal to string.empty to get rid of the error. Lets not have null properties in the product sub classes.

File: ShopAPI/ShopAPI/Models/Product.cs [hunk 1]
```diff
@@ -27,6 +29,47 @@ public class Cpu : Product
...
+    }
+
+    public class Memory : Product
+    {
+        public string MemoryType { get; set; }
+        public int Speed { get; set; }
+        public int Size { get; set; }
+    }
+
+    public class Storage : Product
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Models/Product.cs:53]: Lets add a type for storage (ssd vs hdd) Can keep it simple and have it as a string

**Review by evan-scales — CHANGES_REQUESTED** [review]

File: ShopAPI/ShopAPI/Helpers/ProductHelper.cs [hunk 2]
```diff
@@ -64,17 +64,45 @@ public static class ProductHelper
...
+                                
+                case Category.VideoCard:
+                    var videoCardProperties = typeof(CpuCooler).GetProperties();
+                    requiredProperties = videoCardProperties.Select(p => p.Name);
+                    break;
+                 
+                case Category.PowerSupply:
+                    var powerSupplyProperties = typeof(CpuCooler).GetProperties();
+                    requiredProperties =powerSupplyProperties.Select(p => p.Name);
+                    break;
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Helpers/ProductHelper.cs]: small copy and paste mistake with the typeof()

File: ShopAPI/ShopAPI/Models/Product.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿namespace ShopAPI.Models
+﻿using Microsoft.EntityFrameworkCore.Metadata.Internal;
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Models/Product.cs]: remove this using statement, as its not needed

**Review by evan-scales — CHANGES_REQUESTED** [review]

File: ShopAPI/ShopAPI/Models/Product.cs (hunk 1 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Models/Product.cs:53]: Done

File: ShopAPI/ShopAPI/Helpers/ProductHelper.cs (hunk 2 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Helpers/ProductHelper.cs]: Done

File: ShopAPI/ShopAPI/Models/Product.cs (hunk 3 above)
Comment (epadams) [reply ShopAPI/ShopAPI/Models/Product.cs]: Done

**Review by evan-scales — APPROVED** [review]
looks good

**Pull request merged**